core/src/main/scala/io/finch/request/package.scala

```diff
@@ -162,6 +180,8 @@ package object request {
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) =
         try number.toFuture
         catch { case _: IllegalArgumentException => ValidationFailed(param, rule).toFutureException }
+
+      override def toString: String = s"param: $param"
```

Comment:
That's what I tried to explain in my first message: For single-item readers the reader can provide this info for free (it knows what it is looking at after all), for the more rare cases of a reader for multiple values the user very likely has to fully phrase out something like `should have a startDate smaller than endDate`, where we could prefix a generic "the request". Does that make it clearer?